### PR TITLE
Showcase improvements

### PR DESCRIPTION
**Pull request by wtrocki**
## Technical
Splitting work into smaller PR to simplify review. 
This set of changes are fixing major bugs in current app mostly about error handling.

# What was done

Added gui methods for reporting server side issues:

screen shot 2019-02-14 at 4 16 38 pm

Improved home page
screen shot 2019-02-14 at 4 36 43 pm

Added error handling for tasks
screen shot 2019-02-14 at 5 03 41 pm

Notes: I could not make onlineOnly annotation to work. Logged bug:
https://issues.jboss.org/browse/AEROGEAR-8538

### CODE REVIEW

**Comment by wtrocki** [comment]
Need help on reviewing this changes.
I have applied requirements and suggestions that was made during initial app review. 
Staging this changes separately to simplify review process.

**Comment by wtrocki** [comment]
Note - Fixes are only for non auth enabled case. Working on separate auth page, but wanted to get this changes first.

File: src/app/pages/files/files.page.html [hunk 1]
```diff
@@ -4,7 +4,7 @@
       <ion-menu-button></ion-menu-button>
     </ion-buttons>
     <ion-title>
-      Documents
+      File Management using Voyager Sync
```
Comment (wei-lee) [inline src/app/pages/files/files.page.html]: maybe just call it `Data Sync` as that's the product name?

File: src/app/pages/home/home.page.html [hunk 2]
```diff
@@ -10,41 +10,64 @@
...
+    <ion-card-header>
+      <ion-card-title text-center>AeroGear Mobile Services Showcase Application</ion-card-title>
+    </ion-card-header>
+    <ion-card-content>
+      <ion-label color=stable text-wrap>
+        <div text-center style="padding-top: 2%">
+          <p>
+            This app demonstrates how to use the AeroGear DataSync and other Mobile Services
+            tasks such Data Exchange, Identity Management, Device Security, Metrics Capture, Push Notifications and
+            more.
```
Comment (wei-lee) [inline src/app/pages/home/home.page.html]: can we change this to the origin text we had in the cordova showcase app:

"This app demonstrates how to use the AeroGear Mobile services, backed by OpenShift to perform a wide range of common mobile tasks such as Data Synchronisation, Identity Management, Device Security, Metrics Capture, Push Notifications and more."

File: src/app/pages/home/home.page.html [hunk 3]
```diff
@@ -10,41 +10,64 @@
...
+  </ion-card>
+  <ion-list lines="none">
+    <ion-list-header>
+      <ion-label>Menu</ion-label>
+    </ion-list-header>
+    <ion-item routerDirection='root' routerLink="docs">
+      <ion-icon slot="start" color="medium" name="book"></ion-icon>
+      <ion-label>
+        View Documentation
+        <p>Contains in app documentation for implemented services.</p>
```
Comment (wei-lee) [inline src/app/pages/home/home.page.html]: maybe change this to "AeroGear Mobile Services Documentation"

File: src/app/pages/home/home.page.html [hunk 4]
```diff
@@ -10,41 +10,64 @@
...
+      </ion-label>
+    </ion-item>
+    <ion-item routerDirection='root' routerLink="tasks">
+      <ion-icon slot="start" color="medium" name="build"></ion-icon>
+      <ion-label>
+        Manage Tasks
+        <p>Task manager implemented using Voyager Sync</p>
```
Comment (wei-lee) [inline src/app/pages/home/home.page.html]: maybe change this to something like "An example task manager to demonstrate how to perform data synchronisation with AeroGear Data Sync powered by GraphQL"

File: src/app/pages/home/home.page.html [hunk 5]
```diff
@@ -10,41 +10,64 @@
...
+      </ion-label>
+    </ion-item>
+    <ion-item routerDirection='root' routerLink="profile">
+      <ion-icon slot="start" color="medium" name="person"></ion-icon>
+      <ion-label>User profile
+        <p>Profile page implemented using IDM service</p>
```
Comment (wei-lee) [inline src/app/pages/home/home.page.html]: maybe change this to something like "An example to demonstrate how to perform user profile management with AeroGear Identity Management Service powered by Keycloak"

File: src/app/pages/home/home.page.html [hunk 6]
```diff
@@ -10,41 +10,64 @@
...
+      </ion-label>
+    </ion-item>
+    <ion-item routerDirection='root' routerLink="files">
+      <ion-icon slot="start" color="medium" name="document"></ion-icon>
+      <ion-label>Files
+        <p>File manager using Voyager Sync</p>
```
Comment (wei-lee) [inline src/app/pages/home/home.page.html]: maybe change this to something like "An example to demonstrate how to synchronise binary data with AeroGear Data Sync powered by GraphQL"

File: src/app/pages/home/home.page.html [hunk 7]
```diff
@@ -10,41 +10,64 @@
...
+      </ion-label>
+    </ion-item>
+    <ion-item routerDirection='root' routerLink="security">
+      <ion-icon slot="start" color="medium" name="settings"></ion-icon>
+      <ion-label>Device Security
+        <p>Set of device security checks (supported only on mobile platforms)</p>
```
Comment (wei-lee) [inline src/app/pages/home/home.page.html]: maybe change this to something like "An example to demonstrate push notification management and security checks capabilities powered by AeroGear Push Notification service and Device Security library"

Comment (wtrocki) [reply src/app/pages/home/home.page.html]: I was about to push the changes but rendering seemed off. I think that we agreed to have push in settings page that we going to add soon so removing mention of push from this section. 

ping @ziccardi

**Comment by wtrocki** [comment]
Creating separate PR for auth. @wei-lee Thanks for approval.

**Comment by wtrocki** [comment]
@mmusil This will make you happy :P

**Pull request merged**